a11ym

```diff
@@ -169,54 +175,88 @@ var isErrorOrWarning = function (result) {
     return 'error' === result.type || 'warning' === result.type;
 };
 
-var hasErrors = false;
+var parseURL = function (url) {
+    var parsed = URL.parse(url);
+
+    if (!parsed.protocol) {
+        parsed.protocol = 'http';
+    } else {
+        parsed.protocol = parsed.protocol.replace(':', '');
+    }
+
+    if (!parsed.port) {
+        parsed.port = 'http' === parsed.protocol ? 80 : 443;
+    }
+
+    return parsed;
+};
 
-var urls = async.queue(
+var maximumUrls = +program.maximumUrls;
+var urlQueues = {};
+var hasErrors   = false;
+var isStopped   = false;
+var testQueue   = async.queue(
     function (task, onTaskComplete) {
-        console.log('Run: ' + task.url + '.');
+        if (--maximumUrls < 0) {
+            if (false === isStopped) {
+                isStopped = true;
+                console.log(chalk.white.bgRed('Test queue is stopping, maximum URLs reached.'));
+                testQueue.kill();
+
+                Object
+                    .keys(urlQueues)
+                    .forEach(
+                        function (key) {
+                            urlQueues[key].kill();
+                        }
+                    );
+
+                // In case of emergency.
+                setTimeout(
+                    function () {
+                        process.exit(4);
+                    },
+                    60 * 1000
+                );
+            }
+
+            return;
+        }
+
+        console.log(
+            chalk.black.bgGreen(' ' + (program.maximumUrls - maximumUrls) + '/' + program.maximumUrls + ' ') +
+            ' ' +
+            chalk.black.bgGreen('[' + task.queueName + '] Run: ' + task.url + '.')
+        );
         runTest(
             task.url,
             {
                 complete: function(results) {
+                    task.onUrlComplete();
                     onTaskComplete(null, results);
                 },
                 error: function(error) {
                     hasErrors = true;
+                    task.onUrlComplete();
```

Comment:
See also the comment below, maybe document why we do this. (and the one 4 lines above :P)